lib/src/clojure_lsp/feature/rename.clj

```diff
@@ -225,9 +223,6 @@
                  (not (identical? :namespace-alias (:bucket element)))
                  (not= :rename-file source))
           (let [new-uri (shared/namespace->uri replacement source-paths (:filename definition) db)]
-            (swap! db* (fn [db] (-> db
-                                    (update :documents #(set/rename-keys % {filename (shared/uri->filename new-uri)}))
-                                    (update :analysis #(set/rename-keys % {filename (shared/uri->filename new-uri)})))))
```

Comment:
I know there was some issue with ns renaming when you do it multiple times like: a -> b then b -> a and then a -> b, probably related with this, even so not sure why I added that swap 😅 anyway, this looks correct

Reply:
You mean this was originally a fix for multiple renames, or you suspect it was the source of a problem with multiple renames?

One thing I noticed since writing this PR, is that the rename for `:documents` was actually never working because `:documents` is keyed by uri, not filename.

>  this looks correct

👍 I'll fix conflicts and merge.

Reply:
Yeah, I meant that swap was probably causing that issue when renaming the same file multiple times, it should be fixed now